gobblin-compaction/src/main/java/gobblin/compaction/source/CompactionSource.java

```diff
@@ -177,6 +209,42 @@ public void run () {
     }
   }
 
+  private void initAllocator (State state) {
+    try{
+      Comparator<SimpleDatasetRequest> prioritizer = new SimpleDatasetHierarchicalPrioritizer(state);
+
+      RequestAllocatorConfig.Builder<SimpleDatasetRequest> configBuilder =
+          RequestAllocatorConfig.builder(new SimpleDatasetRequest.SimpleDatasetRequestEstimator()).allowParallelization(1)
+              .withLimitedScopeConfig(ConfigFactory.empty());
+
+        configBuilder.withPrioritizer(prioritizer);
+
+      allocator = new HierarchicalAllocator.Factory().createRequestAllocator(configBuilder.build());
+    } catch (Exception e) {
+      throw new RuntimeException("Cannot initialize allocator");
+    }
+  }
+
+  private List<Dataset> prioritize (List<Dataset> datasets, State state) {
+    double maxPool = state.getPropAsDouble(MRCompactor.COMPACTION_DATASETS_MAX_COUNT, MRCompactor.DEFUALT_COMPACTION_DATASETS_MAX_COUNT);
+    ResourcePool pool = ResourcePool.builder().maxResource(SimpleDatasetRequest.SIMPLE_DATASET_COUNT_DIMENSION, maxPool).build();
+
+    List<Dataset> newList = new ArrayList<>();
+    AllocatedRequestsIterator<SimpleDatasetRequest> allocatedRequests = this.allocator.allocateRequests(Iterators.transform(datasets.iterator(), new Function<Dataset, Requestor<SimpleDatasetRequest>>() {
```

Comment:
With java 8 you can simplify:

`Lists.newArrayList(this.allocator.allocateRequests(datasets.stream().map(SimpleDatasetRequestor::new).iterator(), pool))`